pkg/machine/fedora.go

```diff
@@ -77,21 +80,36 @@ func (f FedoraDownload) CleanCache() error {
 	return removeImageAfterExpire(f.CacheDir, expire)
 }
 
-func getFedoraDownload(releaseURL string) (*url.URL, int64, error) {
+func getFedoraDownload(releaseURL string) (*url.URL, string, int64, error) {
 	downloadURL, err := url.Parse(releaseURL)
 	if err != nil {
-		return nil, -1, fmt.Errorf("invalid URL generated from discovered Fedora file: %s: %w", releaseURL, err)
+		return nil, "", -1, fmt.Errorf("invalid URL generated from discovered Fedora file: %s: %w", releaseURL, err)
 	}
 
 	resp, err := http.Head(releaseURL)
 	if err != nil {
-		return nil, -1, fmt.Errorf("head request failed: %s: %w", releaseURL, err)
+		return nil, "", -1, fmt.Errorf("head request failed: %s: %w", releaseURL, err)
 	}
 	_ = resp.Body.Close()
+	contentLen := resp.ContentLength
 
 	if resp.StatusCode != http.StatusOK {
-		return nil, -1, fmt.Errorf("head request failed: %s: %w", releaseURL, err)
+		return nil, "", -1, fmt.Errorf("head request failed: %s: %w", releaseURL, err)
 	}
 
-	return downloadURL, resp.ContentLength, nil
+	verURL := *downloadURL
+	verURL.Path = path.Join(path.Dir(downloadURL.Path), "version")
+
+	resp, err = http.Get(verURL.String())
+	if err != nil {
+		return nil, "", -1, fmt.Errorf("get request failed: %s: %w", verURL.String(), err)
+	}
+
+	bytes, err := io.ReadAll(&io.LimitedReader{R: resp.Body, N: 1024})
+	if err != nil {
+		return nil, "", -1, fmt.Errorf("failed reading: %s: %w", verURL.String(), err)
+	}
+	_ = resp.Body.Close()
```

Comment:
Doesn't this leak the body when ReadAll fails with an error?

Reply:
yes this is wrong, will fix thanks for the catch